File: app/src/androidTest/java/de/westnordost/streetcomplete/data/osm/persist/NodeDaoTest.java
package de.westnordost.streetcomplete.data.osm.persist;

import java.util.HashMap;
import java.util.Map;

import de.westnordost.streetcomplete.data.ApplicationDbTestCase;
import de.westnordost.osmapi.map.data.LatLon;
import de.westnordost.osmapi.map.data.Node;
import de.westnordost.osmapi.map.data.OsmLatLon;
import de.westnordost.osmapi.map.data.OsmNode;


public class NodeDaoTest extends ApplicationDbTestCase
{
	private NodeDao dao;

	@Override public void setUp() throws Exception
	{
		super.setUp();
		dao = new NodeDao(dbHelper, serializer);
	}

	public void testPutGetNoTags()
	{
		LatLon pos = new OsmLatLon(2,2);
		Node node = new OsmNode(5, 1, pos, null);
		dao.put(node);
		Node dbNode = dao.get(5);

		checkEqual(node, dbNode);
	}

	public void testPutGetWithTags()
	{
		Map<String,String> tags = new HashMap<>();
		tags.put("a key", "a value");
		LatLon pos = new OsmLatLon(2,2);
		Node node = new OsmNode(5, 1, pos, tags);
		dao.put(node);
		Node dbNode = dao.get(5);

		checkEqual(node, dbNode);
	}

	private void checkEqual(Node node, Node dbNode)
	{
		assertEquals(node.getId(), dbNode.getId());
		assertEquals(node.getVersion(), dbNode.getVersion());
		assertEquals(node.getPosition(), dbNode.getPosition());
		assertEquals(node.getTags(), dbNode.getTags());
	}
}

File: app/src/androidTest/java/de/westnordost/streetcomplete/data/osm/persist/NodeDaoTest.kt
package de.westnordost.streetcomplete.data.osm.persist

import org.junit.Before
import org.junit.Test

import de.westnordost.streetcomplete.data.ApplicationDbTestCase
import de.westnordost.osmapi.map.data.Node
import de.westnordost.osmapi.map.data.OsmLatLon
import de.westnordost.osmapi.map.data.OsmNode

import org.junit.Assert.assertEquals


class NodeDaoTest : ApplicationDbTestCase() {
    private lateinit var dao: NodeDao

    @Before fun createDao() {
        dao = NodeDao(dbHelper, NodeMapping(serializer))
    }

    @Test fun putGetNoTags() {
        val pos = OsmLatLon(2.0, 2.0)
        val node = OsmNode(5, 1, pos, null)
        dao.put(node)
        val dbNode = dao.get(5)

        checkEqual(node, dbNode!!)
    }

    @Test fun putGetWithTags() {
        val pos = OsmLatLon(2.0, 2.0)
        val node = OsmNode(5, 1, pos, mapOf("a key" to "a value"))
        dao.put(node)
        val dbNode = dao.get(5)

        checkEqual(node, dbNode!!)
    }

    private fun checkEqual(node: Node, dbNode: Node) {
        assertEquals(node.id, dbNode.id)
        assertEquals(node.version.toLong(), dbNode.version.toLong())
        assertEquals(node.position, dbNode.position)
        assertEquals(node.tags, dbNode.tags)
    }
}
